packages/zoe/src/contracts/callSpread.js

```diff
@@ -0,0 +1,217 @@
+        collateralSeat.stage({ Collateral: collateralRemainder }),
+      );
+      seat.exit();
+    });
+  }
+
+  // calculate the portion (as a percentage) of the collateral that should be
+  // allocated to the long side.
+  function calculateLongShare(price) {
+    if (strikeMath.isGTE(terms.strikePrice1, price)) {
+      return 0;
+    } else if (strikeMath.isGTE(price, terms.strikePrice2)) {
+      return PERCENT_BASE;
+    }
+
+    const denominator = strikeMath.subtract(
+      terms.strikePrice2,
+      terms.strikePrice1,
+    ).value;
+    const numerator = strikeMath.subtract(price, terms.strikePrice1).value;
+    return floorDivide(multiply(PERCENT_BASE, numerator), denominator);
+  }
+
+  function payoffOptions(price) {
+    // either offer might be exercised late, so we pay the two seats separately.
+    const longShare = calculateLongShare(price);
+    reallocateToSeat(Position.LONG, longShare);
+    reallocateToSeat(Position.SHORT, inverse(longShare));
+  }
+
+  function schedulePayoffs() {
+    terms.priceAuthority
+      .priceAtTime(terms.expiration, terms.underlyingAmount)
+      .then(price => payoffOptions(price));
+  }
+
+  function makeOptionInvitation(dir) {
+    const optionsTerms = harden({
+      ...terms,
+      position: dir,
+    });
+    // All we do at time of exercise is resolve the promise.
+    return zcf.makeInvitation(
+      seat => seatPromiseKits[dir].resolve(seat),
+      `collect ${dir} payout`,
+      optionsTerms,
+    );
+  }
+
+  async function makeOptionPair() {
+    return {
+      longInvitation: makeOptionInvitation(Position.LONG),
+      shortInvitation: makeOptionInvitation(Position.SHORT),
+    };
+  }
+
+  async function makeInvitationToBuy() {
+    const { longInvitation, shortInvitation } = await makeOptionPair();
+    const invitationIssuer = zcf.getInvitationIssuer();
+    const longAmount = await E(invitationIssuer).getAmountOf(longInvitation);
+    const shortAmount = await E(invitationIssuer).getAmountOf(shortInvitation);
+    depositToSeat(
+      zcf,
+      collateralSeat,
+      { LongOption: longAmount, ShortOption: shortAmount },
+      { LongOption: longInvitation, ShortOption: shortInvitation },
+    );
+
+    // transfer collateral from longSeat to collateralSeat, then return a pair
+    // of callSpread invitations
+    /** @type {OfferHandler} */
+    const pairBuyerPosition = longSeat => {
+      assertProposalShape(longSeat, {
+        give: { Collateral: null },
+        want: { LongOption: null, ShortOption: null },
+      });
+
+      trade(
+        zcf,
+        {
+          seat: collateralSeat,
+          gains: { Collateral: terms.settlementAmount },
+        },
+        {
+          seat: longSeat,
+          gains: { LongOption: longAmount, ShortOption: shortAmount },
+        },
+      );
+      schedulePayoffs();
+      longSeat.exit();
+    };
+
+    const longTerms = harden({
+      ...terms,
+      LongOption: longAmount,
+      ShortOption: shortAmount,
```

Comment:
Let's not use capitalization in the customProperties. Can we call this 
```suggestion
      longAmount,
      shortAmount,
```
instead?

Reply:
done